modules/Core/classes/Integrations/MinecraftIntegration.php

```diff
@@ -0,0 +1,190 @@
+<?php
+
+/*
+ *	Made by Partydragen
+ *  https://github.com/NamelessMC/Nameless/
+ *  NamelessMC version 2.0.0-pr13
+ *
+ *  License: MIT
+ *
+ *  Minecraft Integration
+ */
+
+class MinecraftIntegration extends IntegrationBase {
+
+    protected Language $_language;
+    private string $_uuid;
+
+    public function __construct(Language $language) {
+        $this->_name = 'Minecraft';
+        $this->_icon = 'fas fa-cubes';
+        $this->_language = $language;
+
+        parent::__construct();
+    }
+
+    public function onLinkRequest(User $user) {
+        $queries = new Queries();
+        $username = $user->data()->username;
+
+        // Ensure username doesn't already exist
+        $integrationUser = new IntegrationUser($this, $username, 'username');
+        if ($integrationUser->exists()) {
+            $this->addError(str_replace('{x}', $this->getName(), $this->_language->get('user', 'integration_username_already_linked')));
+            return;
+        }
+
+        $result = $this->getUuidByUsername($username);
+        if (count($this->getErrors())) {
+            return;
+        }
+        $this->_uuid = $result['uuid'];
+
+        // Ensure identifier doesn't already exist
+        $integrationUser = new IntegrationUser($this, $this->_uuid, 'identifier');
+        if ($integrationUser->exists()) {
+            $this->addError(str_replace('{x}', $this->getName(), $this->_language->get('user', 'integration_identifier_already_linked')));
+            return;
+        }
+
+        $code = substr(str_shuffle('0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ'), 0, 10);
+
+        $integrationUser = new IntegrationUser($this);
+        $integrationUser->linkIntegration($user, $this->_uuid, $username, false, $code);
+
+        // API verification
+        $api_verification = $queries->getWhere('settings', ['name', '=', 'api_verification']);
+        $api_verification = $api_verification[0]->value;
+
+        if ($api_verification == '1') {
+            Session::flash('connections_success', str_replace('{x}', Output::getClean($code), $this->_language->get('user', 'validate_account_command')));
+        } else {
+            Session::flash('connections_success', str_replace('{x}', Output::getClean($this->getName()), $this->_language->get('user', 'integration_linked')));
+        }
+    }
+
+    public function onVerifyRequest(User $user) {
```

Comment:
i'm confused what the difference between `onVerifyRequest` and `onLinkRequest` is... do they not verify that they own the account when they complete the linking process?

Reply:
No, link does not need to verify as Minecraft and Discord is pending once they execute command